### PR TITLE
[helper-plugin]: migrate icons folder to typescript

### PR DESCRIPTION
**Pull request by madhurisandbhor**
### What does it do?

Migrate helper-plugin/icons to typescript.

### Related issue(s)/PR(s)

Merge after DS fix: https://github.com/strapi/design-system/pull/1306
Main Issue: https://github.com/strapi/strapi/issues/17690

### CODE REVIEW

File: packages/core/helper-plugin/src/icons/RemoveRoundedButton.tsx [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+import { IconButton, IconButtonProps } from '@strapi/design-system';
 import { Minus } from '@strapi/icons';
-import styled from 'styled-components';
+import styled, { DefaultTheme } from 'styled-components';
 
 import { pxToRem } from '../utils/pxToRem';
 
 const StyledIconButton = styled(IconButton)(
-  ({ theme }) => `
+  ({ theme }: { theme: DefaultTheme }) => `
```
Comment (joshuaellis) [inline packages/core/helper-plugin/src/icons/RemoveRoundedButton.tsx:10]: I don't think you need this, the `.d.ts` file should augment the global scope of the package, was this a legacy?

File: packages/core/helper-plugin/src/icons/SortIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+
+import { CarretDown } from '@strapi/icons';
+import styled from 'styled-components';
+
+export interface SortIconProps extends React.SVGProps<SVGSVGElement> {
+  isUp: boolean;
+}
+
+const transientProps: Partial<Record<keyof SortIconProps, boolean>> = {
```
Comment (markkaylor) [inline packages/core/helper-plugin/src/icons/SortIcon.tsx:10]: Couldn't we just let TypeScript infer this type? It feels a bit verbose for the one key we are using but maybe there is something I'm missing.

**Review by markkaylor — COMMENTED** [review]
Looks good to me I just have one question

File: packages/core/helper-plugin/src/icons/SortIcon.tsx (hunk 2 above)
Comment (madhurisandbhor) [reply packages/core/helper-plugin/src/icons/SortIcon.tsx:10]: hmm...we are referring it as `transientProps[prop]` , and typescript suggests `(parameter) prop: keyof SortIconProps`. So we need to define it explicitly as above.

Comment (markkaylor) [reply packages/core/helper-plugin/src/icons/SortIcon.tsx:10]: ok works for me :)

**Review by joshuaellis — APPROVED** [review]

**Review by markkaylor — DISMISSED** [review]
Just not sure what's up with that e2e test timing out 🤔

File: packages/core/helper-plugin/src/icons/SortIcon.jsx
Comment (dallasclark) [inline packages/core/helper-plugin/src/icons/SortIcon.jsx:1]: Ideally you would want to move a file using Git so you can retain all of the file's history before and after the filename changes.

**Review by joshuaellis — APPROVED** [review]

**Pull request merged**